### PR TITLE
ar71xx: Add support for D-Link DAP-1330 a1

### PR DESCRIPTION
**Pull request by hitech95**
D-Link DAP-1330 rev. A1 is a wall-plug N300 Wi-Fi range extender,
based on Qualcomm/Atheros QCA9533 v2.

Short specification:

- 650/393/216 MHz (CPU/DDR/AHB)
- 1x 10/100 Mbps Ethernet
- 64 MB of RAM (DDR1)
- 8 MB of FLASH
- 2T2R 2.4 GHz
- 2x external antennas
- 6x LED (2 are bi-color), 2x button
- 4 pin on pcb (looking from eth port and from left to right tx,rx,gnd,vcc)

Flash instruction: use "factory" image directly in vendor GUI.

This device has a recovery system that assign the ip addr of env
variable "serverip" via dhcp to a pc, and the "ipaddr" value to itself.
The recovery it's triggered by a not bootable firmware,
or pressing the reset button during the bootloader startup (first 30 seconds).
The recovery uses a http page to restore the firmware, and it's checking
the firmware image header, so use the "factory" image to
restore or the original firmware.

The default recovery IP is: 192.168.1.1

You can access vendor firmware over serial using:
- login: root
- password: linuxrocks

Image was tested only in EU version of the device, but should work
also with the same device version sold in other countries.

For more info look at lede-project/source#1177

### CODE REVIEW

File: target/linux/ar71xx/image/generic.mk [hunk 1]
```diff
@@ -32,6 +32,16 @@ define Build/mkwrggimg
 	mv $@.imghdr $@
 endef
 
+define Build/mkdapimg2
+	$(STAGING_DIR_HOST)/bin/mkdapimg2 \
+		-i $@ -o $@.new \
+		-s $(DAP_SIGNATURE) \
+		-v LEDE-$(firstword $(subst -, ,$(REVISION))) \
```
Comment (NeoRaider) [inline target/linux/ar71xx/image/generic.mk]: Change "LEDE" to "OpenWrt" here.

**Review by neocturne — CHANGES_REQUESTED** [review]
Apart from one nitpick, LGTM.

File: target/linux/ar71xx/image/generic.mk (hunk 1 above)
Comment (hitech95) [reply target/linux/ar71xx/image/generic.mk]: Ops :) Missed that!

Comment (mkresin) [reply target/linux/ar71xx/image/generic.mk]: Even better would be if you use the $(VERSION_DIST) variable. Have a look at https://git.openwrt.org/43b432050c73a233a89012f13f01e81de1629611 for reference.

Comment (KanjiMonster) [reply target/linux/ar71xx/image/generic.mk]: Actually, use `$(VERSION_DIST)`

Comment (hitech95) [reply target/linux/ar71xx/image/generic.mk]: Done :)

**Comment by bobafetthotmail** [comment]
Another minor issue:

What is the IP address of the recovery http page you mention? Please add that to the commit description.

**Comment by hitech95** [comment]
@bobafetthotmail 
If I remember correctly the IP is auto generated (pseudorandom) by the uboot when you write the commsnds... so it is quite annoing. You can print the env and see the value from there.

I worked on this patch like 8 month ago I don't remember each detail of the hardware. Now i can flash it via the LUCI web interface...

**Comment by bobafetthotmail** [comment]
>If I remember correctly the IP is auto generated (pseudorandom) by the uboot when you write the commsnds... so it is quite annoing. You can print the env and see the value from there.

I think I was not clear. If I have serial console I can use printenv as you said.

I meant this part, where the user does not have serial console but there is a http page:

> The recovery it's triggered by a not bootable firmware,
> or pressing the reset button during the bootloader startup (first 30 seconds).
> The recovery uses a http page to restore the firmware

What is the IP of the webpage? It should have a static IP afaik.
From here http://forums.dlink.com/index.php?topic=58357.0 it seems the recovery webpage IP for a similar device is 192.168.0.50 and from googling it seems D-link http recovery seems to have IP in that subnet (so from 192.168.0.0 to 192.168.0.255), so you can just try with nmap/Zenmap "ping scan" in that subnet to find them.

Can you please test this and add it to the commit description?

**Comment by hitech95** [comment]
@NeoRaider Any news?

**Comment by NeoRaider** [comment]
Thanks, merged.

When rebasing I noticed two more issues which I fixed up for you: There was a `;;` missing in one of the scripts, and DEVICE_VARS must not be modified inside a device definition.

**Pull request closed without merging**